docs/source/serving/https.rst

```diff
@@ -0,0 +1,74 @@
+.. _https:
+
+HTTPS Encryption
+================
+
+SkyServe enables secure serving of models over HTTPS, which is essential for handling sensitive data or for models that require secure communication with other services. Currently, SkyServe supports HTTPS only on the load balancer (for communication between the client and the load balancer); HTTPS on the service replicas (for communication between the load balancer and the service replicas) is not yet supported.
+
+HTTPS on Load Balancer
+----------------------
+
+To enable HTTPS on the load balancer, you need to provide a certificate and a private key. Obtaining these from a trusted Certificate Authority (CA) is the most secure method. However, for development and testing purposes, you can generate a self-signed certificate and private key using the :code:`openssl` command-line tool. Here is an example of how to generate them:
```

Comment:
We should have a section talking about common ways to get certificate for people who want to do it in production.

Reply:
Also, does a normal HTTPS requires periodic refresh for the token?

Reply:
Good point! Lemme investigate and try it out.

Reply:
Yes, they typically last for 3 months and [up to 13 months](https://www.digicert.com/faq/public-trust-and-certificates/how-long-are-tls-ssl-certificate-validity-periods). FastAPI doc says you cannot do that using uvicorn (see picture below). I think moving to envoy would address this problem.

https://fastapi.tiangolo.com/deployment/https/#certificate-renewal

<img width="751" alt="image" src="https://github.com/user-attachments/assets/6f8bd729-cd77-4640-8c0a-51ae39e08692">